### PR TITLE
[PRM-1141] Fixed incorrect routing on non-js evidence upload

### PR DESCRIPTION
**Pull request by ruthdas03**
https://user-images.githubusercontent.com/86231805/157640597-b1e4b840-0e5b-4e3b-acbd-9fca77660377.mov

### CODE REVIEW

File: app/controllers/OtherReasonController.scala [hunk 1]
```diff
@@ -128,14 +128,11 @@ class OtherReasonController @Inject()(whenDidBecomeUnablePage: WhenDidBecomeUnab
...
-            case CheckMode => if (previousUploadsState.exists(_.count(_.fileStatus == READY) > 0)) {
-              Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
-            } else {
-              Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
-            }
-          }
+          val uploadState: Boolean = previousUploadsState.exists(_.count(_.fileStatus == READY) > 0)
+          (mode, uploadState) match {
+            case (NormalMode | CheckMode, true) => Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
+            case (NormalMode | CheckMode, false) => Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
```
Comment (LewisMitchell) [inline app/controllers/OtherReasonController.scala:134]: I'm just thinking, do we now need to match on the `mode`? Could we do something like this:
 Rename uploadState to hasReadyUploads and have an if statement like: 
```
if(hasReadyUploads) Redirect(controllers.routes.OtherReasonController.onPageLoadForUploadComplete(mode))
else Redirect(controllers.routes.OtherReasonController.onPageLoadForFirstFileUpload(mode))
```

File: app/controllers/OtherReasonController.scala [hunk 2]
```diff
@@ -362,11 +359,11 @@ class OtherReasonController @Inject()(whenDidBecomeUnablePage: WhenDidBecomeUnab
...
-        logger.debug(
-          s"[OtherReasonController][onSubmitForUploadEvidenceQuestion] - Adding '$isUploadEvidenceQuestion' to session under key: ${SessionKeys.isUploadEvidence}")
-        Redirect(navigation.nextPage(UploadEvidenceQuestionPage, mode, Some(isUploadEvidenceQuestion)))
-          .addingToSession((SessionKeys.isUploadEvidence, isUploadEvidenceQuestion))
-      }
+            logger.debug(
+              s"[OtherReasonController][onSubmitForUploadEvidenceQuestion] - Adding '$isUploadEvidenceQuestion' to session under key: ${SessionKeys.isUploadEvidence}")
+            Redirect(navigation.nextPage(UploadEvidenceQuestionPage, mode, Some(isUploadEvidenceQuestion)))
+              .addingToSession((SessionKeys.isUploadEvidence, isUploadEvidenceQuestion))
+          }
```
Comment (LewisMitchell) [inline app/controllers/OtherReasonController.scala:366]: Could you remove the excess indentation?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor changes. Yet to run AT's

File: app/controllers/OtherReasonController.scala (hunk 2 above)
Comment (ruthdas03) [reply app/controllers/OtherReasonController.scala]: Yes 😃 , done

File: app/controllers/OtherReasonController.scala (hunk 1 above)
Comment (ruthdas03) [reply app/controllers/OtherReasonController.scala]: 👍

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**